gnomad_qc/v4/sample_qc/identify_trios.py

```diff
@@ -0,0 +1,482 @@
+    get_duplicated_samples,
+    get_duplicated_samples_ht,
+    infer_families,
+)
+from gnomad.utils.slack import slack_notifications
+from gnomad.utils.vcf import SEXES
+from hail.utils.misc import new_temp_file
+
+from gnomad_qc.resource_utils import (
+    PipelineResourceCollection,
+    PipelineStepResourceCollection,
+)
+from gnomad_qc.slack_creds import slack_token
+from gnomad_qc.v4.resources.basics import get_gnomad_v4_vds
+from gnomad_qc.v4.resources.sample_qc import (
+    duplicates,
+    finalized_outlier_filtering,
+    ped_mendel_errors,
+    pedigree,
+    relatedness,
+    sample_rankings,
+    sex,
+    trios,
+)
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("identify_trios")
+logger.setLevel(logging.INFO)
+
+
+def families_to_trios(ped: hl.Pedigree) -> hl.Pedigree:
+    """
+    Convert a pedigree with families to a pedigree with only one trio per family.
+
+    :param ped: Pedigree with families.
+    :return: Pedigree with only one trio per family.
+    """
+    trios_per_fam = defaultdict(list)
+    for trio in ped.trios:
+        trios_per_fam[trio.fam_id].append(trio)
+
+    message = [
+        f"Found a total of {len(ped.trios)} trios in {len(trios_per_fam)} families:"
+    ]
+    for n_trios, n_fam in Counter([len(t) for t in trios_per_fam.values()]).items():
+        message.append(f"{n_fam} with {n_trios} trios.")
+    logger.info("\n".join(message))
+
+    return hl.Pedigree(trios=[t[0] for t in trios_per_fam.values()])
+
+
+def filter_relatedness_ht(ht: hl.Table, filter_ht: hl.Table) -> hl.Table:
+    """
+    Filter relatedness Table to only include pairs of samples that are both exomes and not QC-filtered.
+
+    :param ht: Relatedness Table.
+    :param filter_ht: Outlier filtering Table.
+    :return: Filtered relatedness Table.
+    """
+    ht = ht.filter((ht.i.data_type == "exomes") & (ht.j.data_type == "exomes"))
+    ht = ht.key_by(i=ht.i.s, j=ht.j.s)
+
+    # Remove all pairs with a QC-filtered sample
+    ht = ht.filter(
+        filter_ht[ht.i].outlier_filtered | filter_ht[ht.j].outlier_filtered,
+        keep=False,
+    )
+
+    return ht
+
+
+def run_create_fake_pedigree(
+    ped: hl.Pedigree, filter_ht: hl.Table, fake_fam_prop: float = 0.1
+) -> hl.Pedigree:
+    """
+    Generate a fake pedigree with `fake_fam_prop` of the number of trios in `ped`.
+
+    :param ped: Pedigree to use for generating fake pedigree.
+    :param filter_ht: Outlier filtering Table.
+    :param fake_fam_prop: Proportion of trios in `ped` to use for generating fake
+        pedigree. Default is 0.1.
+    :return: Fake pedigree.
+    """
+    n_fake_trios = int(fake_fam_prop * len(ped.complete_trios()))
+    logger.info("Generating fake pedigree with %i trios.", n_fake_trios)
+    fake_ped = create_fake_pedigree(
+        n=n_fake_trios,
+        sample_list=list(filter_ht.filter(filter_ht.outlier_filtered).s.collect()),
+        real_pedigree=ped,
+    )
```

Comment:
If you don't want there to be any overlap of probands in the real pedigree and the fake pedigree. For our purpose I don't think it matters that much, we could use either option, it still makes sure we don't have real trios in the fake dataset which is most important. I'm fine adding `exclude_real_probands=True` if you think it would be better